### DISCUSSION TITLE
Javascript regex undefined results in REST transformer

### DISCUSSION THREAD

**Discussion by rourke** [Budibase/budibase#5894] [Help]
I've created a REST api call towards Paperless-ngx to fetch some metadata of specific monthly invoices I receive. I use a transformer to regex the content of the invoice to get the information I want. The transformer looks like this:

```javascript
let returnData = [];
for (let key in data.results) {
    const invoice = data.results[key];
    let returnRow = { paperless_id: invoice.id };

    const regex = /(\d{4}-\d{2}).*Factuurnummer:.*(\d{10,12}).*Debiteurnummer:.*(\d{8,10})[^€]*€[^€]*€(\d.\d{1,3},\d{2})/gs;

    let matches = regex.exec(invoice.content);

    returnRow.matches = matches;

    returnData.push(returnRow);
}
return returnData;
```

The content of an example invoice I'm trying to extract data from, with all sensitive information removed or replaced with random data:

```
Name\nAddress\nAddress\n\n\nInvoice\n\n2022-05\n\nFactuurnummer: 2022871237\nPeriode:\nDebiteurnummer: 82819928\nFactuurdatum:\nVervaldatum:\n\n22-04-2022\n06-05-2022\n\nSpecificatie\n\nFirst name Last Name\n\n€1.124,10\n€1.124,10\n\ntype \nCompany name,\nCompany name,\nMetadata\n01-02-2022\n09-08-2024\nCompany name\nCompany name\nMetadata\n\n\n\n\n\nContract information\nPrice according to contract\n\nPeriod time\n\n\n\n\n\n\n\nX\nX
```

This returns:

```json
{
  "paperless_id": 113,
  "matches": [
    "2022-05\n\nFactuurnummer: 2022871237\nPeriode:\nDebiteurnummer: 82819928\nFactuurdatum:\nVervaldatum:\n\n22-04-2022\n06-05-2022\n\nSpecificatie\n\nFirst name Last Name\n\n€1.124,10\n€1.124,10",
    "2022-05",
    "2022871237",
    "82819928",
    "1.124,10"
  ]
}
```

So I would expect to be able to fetch `matches[1]` until `matches[4]`, but it returns an error `Query Error: Cannot read property '1' of null`. When I run this locally, or for example on JsFiddle it works: https://jsfiddle.net/kutwo68n/.

I'm not sure what is going on. I've been looking at it for hours. So hopefully someone much brighter can help me out.

**Comment by melohagan** [comment]
Hi @rourke 

What does your schema look like for your REST query? Where are you trying to fetch `matches[1]` ?

The payload shown in the **JSON** tab is an example **row**:
![Screenshot 2022-05-16 at 15 54 37](https://user-images.githubusercontent.com/101575380/168621859-d0cd4344-d30b-4ffe-a950-92c7848f6eeb.png)
